libs/template/config.go

```diff
@@ -127,11 +127,33 @@ func (c *config) skipPrompt(p jsonschema.Property, r *renderer) (bool, error) {
 
 	// Check if conditions specified by template author for skipping the prompt
 	// are satisfied. If they are not, we have to prompt for a user input.
-	for name, property := range p.Schema.SkipPromptIf.Properties {
-		if v, ok := c.values[name]; ok && v == property.Const {
-			continue
+	if p.Schema.SkipPromptIf.Properties != nil {
+		for name, property := range p.Schema.SkipPromptIf.Properties {
+			if v, ok := c.values[name]; ok && v == property.Const {
+				continue
+			}
+			return false, nil
+		}
+	}
+
+	// Check if AnyOf is set first. If so, then iterate over all property sets to find a match.
+	// If no match is found (allFalse stays true), we have to prompt for a user input.
+	if p.Schema.SkipPromptIf.AnyOf != nil {
```

Comment:
This section is hard to read because of multiple levels of indentation and the boolean logic. Could we refactor this validation for `anyOf` and `const` into their own functions? and also test it separately? Something similar to: https://github.com/databricks/cli/blob/7067782cf165e3a18ffd24009a28aeb57e20f5c1/libs/jsonschema/schema.go#L231

We can return early once any validation rules fail.

Reply:
Infact we already have a function to validate a config against JSON schema. Could you add the rules for `const` and `anyOf` to the ValidateInstance function and use them here?

https://github.com/databricks/cli/blob/7067782cf165e3a18ffd24009a28aeb57e20f5c1/libs/jsonschema/instance.go#L43

Reply:
Sounds good refactored accordingly